monai/transforms/intensity/dictionary.py

```diff
@@ -1020,6 +1017,70 @@ def __call__(self, data: Mapping[Hashable, np.ndarray]) -> Dict[Hashable, np.nda
         return d
 
 
+class RandGibbsNoised(RandomizableTransform, MapTransform):
+    """
+    Dictionary-based version of RandGibbsNoise.
+
+    Naturalistic image augmentation via Gibbs artifacts. The transform 
+    randomly applies Gibbs noise to 3D MRI images. Gibbs artifacts
+    are one of the common type of type artfifacts appearing in MRI scans.
+    
+    For general information on Gibbs artifacts, please refer to:
+    https://pubs.rsna.org/doi/full/10.1148/rg.313105115
+    https://pubs.rsna.org/doi/full/10.1148/radiographics.22.4.g02jl14949
+    
+    Args:
+        prob: probability of applying the transform.
+        keys: 'image', 'label', or ['image', 'label'] depending on which data
+                you need to transform.
+        alpha: Parametrizes the intensity of the Gibbs noise filter applied. Smaller
+            values of alpha correspond to milder applications. If a length-2 list is 
+            given as [a,b] then the value of alpha will be sampled uniformly from the 
+            interval [a,b]. 
+        allow_missing_keys: don't raise exception if key is missing.
+
+    """
+
+    def __init__(
+        self,
+        keys: KeysCollection,
+        prob: float = 0.1,
+        alpha: Union[float, List[float]] = 0.5,
+        allow_missing_keys: bool = False,
+    ) -> None:
+
+        if type(alpha) == list:
+            assert alpha[1] <= 1 and alpha[0] >= 0, "alpha must take values in [0,1]."
+            assert alpha[0] <= alpha[1], "When alpha = [a,b] we need a < b."
+            assert len(alpha) == 2, "If alpha is a list, then its length must be 2."
+
+        self.alpha = alpha
+
+        MapTransform.__init__(self, keys, allow_missing_keys)
+        RandomizableTransform.__init__(self, prob=prob)
+
+    def __call__(self, data: Mapping[Hashable, torch.tensor]) -> Dict[Hashable, torch.tensor]:
+
+        d = dict(data)
+        # randomize application and possibly alpha
+        self.randomize()
+        transform = GibbsNoise(self.alpha)
+
+        if self._do_transform:
+            for key in self.key_iterator(d):
+                d[key] = transform(d[key])
+        return d
+
+    def randomize(self) -> None:
+        """
+        (1) Set random variable to apply the transform.
+        (2) Get alpha from uniform distribution.
+        """
+        super().randomize(None)
+        if type(self.alpha) == list:
+            self.alpha = self.R.uniform(self.alpha[0], self.alpha[1])
```

Comment:
since you overwrite `self.alpha`, the randomisation will only work the first time. How about 

```suggestion
        return self.R.uniform(self.alpha[0], self.alpha[1]) if isinstance(self.alpha, list) else self.alpha
```

and then you call it with

```python
alpha = self.randomize()
transform = GibbsNoise(alpha)
```

You'd need to do the same for the array version